apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -427,17 +454,13 @@ export class RushConfiguration {
       }
 
       const knownSet: Set<string> = new Set<string>(knownRushConfigFilenames.map(x => x.toUpperCase()));
-      switch (packageManager) {
-        case 'npm':
-          knownSet.add(RushConstants.npmShrinkwrapFilename.toUpperCase());
-          break;
-        case 'pnpm':
-          knownSet.add(RushConstants.pnpmShrinkwrapFilename.toUpperCase());
-          knownSet.add(RushConstants.pnpmfileFilename.toUpperCase());
-          break;
-        case 'yarn':
-          knownSet.add(RushConstants.yarnShrinkwrapFilename.toUpperCase());
-          break;
+
+      // Add the shrinkwrap filename for the package manager to the known set.
+      knownSet.add(shrinkwrapFilename.toUpperCase());
+
+      // If the package manager is pnpm, then also add the pnpm file to the known set.
+      if (packageManager === 'pnpm') {
+        knownSet.add(RushConstants.pnpmfileFilename.toUpperCase());
```

Comment:
Maybe the known files should come from a `PackageManagerFeatureSet` object, which could get passed into this function.

Reply:
We discussed this and decided that `PackageManagerFeatureSet` should be renamed to `PackageManager` and planned to eventually consolidate all of the package-manager-specific operations (but not in this PR).  This will be a breaking change with the current `PackageManager` type alias, but we doubted that anyone is relying on that (and I pointed out that it's a comple-time break but not a runtime break).